assets/js/modules/analytics-4/utils/withCustomDimensions.test.js

```diff
@@ -89,6 +98,53 @@ describe( 'withCustomDimensions', () => {
 		expect( container ).toHaveTextContent( 'Insufficient permissions' );
 	} );
 
+	it( 'sets appropriate `redirectURL` in persmission error object if creating custom dimensions failed due to the user not having `EDIT_SCOPE`', () => {
+		registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
+			propertyID: '123456789',
+			availableCustomDimensions: [ customDimension ],
+		} );
+
+		provideUserAuthentication( registry, {
+			grantedScopes: [],
+		} );
+
+		const error = {
+			code: 'test-error-code',
+			message: 'Test error message',
+			data: {
+				reason: ERROR_REASON_INSUFFICIENT_PERMISSIONS,
+			},
+		};
+
+		provideCustomDimensionError( registry, {
+			customDimension,
+			error,
+		} );
+
+		const { queryByText, getByRole } = render(
+			<WithCustomDimensionsComponent />,
+			{
+				registry,
+			}
+		);
+
+		expect( queryByText( /retry/i ) ).toBeInTheDocument();
+
+		fireEvent.click( getByRole( 'button', { name: /retry/i } ) );
+
+		const redirectURL = addQueryArgs( global.location.href, {
+			notification: 'custom_dimensions',
+		} );
+
+		const permissionScopeError = registry
+			.select( CORE_USER )
+			.getPermissionScopeError();
+
+		expect( permissionScopeError?.data?.redirectURL ).toMatch(
```

Comment:
I would imagine this conditional chaining isn't needed as we expect `redirectURL` to exist at this point:

```suggestion
		expect( permissionScopeError.data.redirectURL ).toMatch(
```